src/framework/theme/components/select/option-group.component.ts

```diff
@@ -26,13 +40,49 @@ export class NbOptionGroupComponent {
   }
   set disabled(value: boolean) {
     this._disabled = convertToBoolProperty(value);
+
+    if (this.options) {
```

Comment:
Don't you think we have to check `this.options.length` here?
```suggestion
    if (this.options && this.options.length) {
```

Reply:
I don't see why we should add additional check. In `updateOptionsDisabledState` we iterate through options and if there is no options, nothing will be done.